rust/sealevel/client/src/router.rs

```diff
@@ -555,6 +566,82 @@ fn configure_owner(
     }
 }
 
+/// Idempotent. Attempts to set the upgrade authority to the intended owner if
+/// the payer can change the upgrade authority.
+fn configure_upgrade_authority(
+    ctx: &mut Context,
+    program_id: &Pubkey,
+    router_config: &RouterConfig,
+    chain_config: &ChainMetadata,
+) {
+    let client = chain_config.client();
+
+    let program_account = client.get_account(program_id).unwrap();
+    // If the program isn't upgradeable, exit
+    if program_account.owner == bpf_loader_upgradeable::id() {
+        return;
+    }
+
+    // The program id must actually be a program
+    let programdata_address = if let Ok(UpgradeableLoaderState::Program {
+        programdata_address,
+    }) = program_account.state()
+    {
+        programdata_address
+    } else {
+        return;
```

Comment:
nit: can we have an UpgradeWarning enum we return with the specified reasons here, or is it a bit unnecessary?

Reply:
Imo the quality bar for the SVM tooling is far below that 😅